RequestService/RequestService.Repo/Repository.cs

```diff
@@ -1136,11 +1138,25 @@ public int UpdateShiftStatusToAccepted(int requestID, HelpMyStreet.Utils.Enums.S
 
             if (job != null)
             {
-                job.JobStatusId = (byte)JobStatuses.Accepted;
-                job.VolunteerUserId = volunteerUserID;
-                AddJobStatus(job.Id, createdByUserID, volunteerUserID, (byte)JobStatuses.Accepted);
-                _context.SaveChanges();
-                return job.Id;
+                //final check before adding volunteer to make sure that for given requestID and support activity
+                //that the volunteer has not accepted a shift
+                int count = _context.Job.Count(x => x.RequestId == requestID
```

Comment:
To make this totally threadsafe, we neeed to move the check to _after_ AddJobStatus(job.Id, createdByUserID, volunteerUserID, (byte)JobStatuses.Accepted);, if not after _context.SaveChanges();

At that point, we should check count < 2 and only roll back change if not.